src/objective-c/tests/BUILD

```diff
@@ -0,0 +1,235 @@
+# gRPC Bazel BUILD file.
+#
+# Copyright 2019 gRPC authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+licenses(["notice"])  # Apache v2
+
+package(default_visibility = ["//visibility:private"])
+
+load(
+    "//src/objective-c:grpc_objc_internal_library.bzl",
+    "grpc_objc_testing_library",
+    "testing_objc_grpc_library"
+)
+load("@build_bazel_rules_apple//apple:resources.bzl", "apple_resource_bundle")
+load("@build_bazel_rules_apple//apple:ios.bzl", "ios_application", "ios_unit_test")
+load("@build_bazel_rules_apple//apple:macos.bzl", "macos_unit_test")
+load("@build_bazel_rules_apple//apple:tvos.bzl", "tvos_application", "tvos_unit_test")
+
+exports_files(["LICENSE"])
+
+proto_library(
+    name = "messages_proto",
+    srcs = ["RemoteTestClient/messages.proto"],
+)
+
+proto_library(
+    name = "test_proto",
+    srcs = ["RemoteTestClient/test.proto"],
+    deps = [":messages_proto"],
+)
+
+testing_objc_grpc_library(
+    name = "RemoteTest",
+    srcs = ["RemoteTestClient/test.proto"],
+    use_well_known_protos = True,
+    deps = [":test_proto"],
+)
+
+apple_resource_bundle(
+    name = "TestCertificates",
+    resources = ["TestCertificates.bundle/test-certificates.pem"],
+)
+
+# TestConfigs is added to each grpc_objc_testing_library's deps
+grpc_objc_testing_library(
+    name = "TestConfigs",
+    hdrs = ["version.h"],
+    data = [":TestCertificates"],
+    defines = [
+        "DEBUG=1",
+        "PB_FIELD_32BIT=1",
+        "PB_NO_PACKED_STRUCTS=1",
+        "PB_ENABLE_MALLOC=1",
+        "HOST_PORT_LOCALSSL=localhost:5051",
+        "HOST_PORT_LOCAL=localhost:5050",
+        "HOST_PORT_REMOTE=grpc-test.sandbox.googleapis.com",
+    ],
+)
+
+objc_library(
+    name = "host-lib",
+    srcs = glob(["Hosts/ios-host/*.m"]),
+    hdrs = glob(["Hosts/ios-host/*.h"]),
+)
+
+ios_application(
+    name = "ios-host",
```

Comment:
why do we need an iOS host?

Reply:
Because running without it doesn't work when making requests (going through the entire network stack) to the sandbox server. Haven't figured out why not tho.